### PR TITLE
Added implementation

### PR DESCRIPTION
**Pull request by oleksandrdziubiuk**

### CODE REVIEW

**Review by engineer027 — APPROVED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (SessionException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new SessionException("Can't insert Movie entity", e);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
            throw new DataProcessingException("Can't insert Movie entity " + movie, e);
```

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class MovieDaoImpl implements MovieDao {
+
+    @Override
+    public Movie add(Movie movie) throws SessionException {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]:
```suggestion
    public Movie add(Movie movie) {
```

File: src/main/java/com/dev/cinema/exception/SessionException.java [hunk 3]
```diff
@@ -0,0 +1,7 @@
+package com.dev.cinema.exception;
+
+public class SessionException extends RuntimeException {
+    public SessionException(String message, SessionException e) {
+        super(message);
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/exception/SessionException.java]:
```suggestion
        super(message, e);
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    public Movie add(Movie movie) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.save(movie);
+            transaction.commit();
+            return movie;
+        } catch (DataProcessingException e) {
```
Comment (RomanDubovskyi) [inline src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: wrote in channel about it

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java (hunk 4 above)
Comment (oleksandrdziubiuk) [reply src/main/java/com/dev/cinema/dao/impl/MovieDaoImpl.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**